lib/crystalball/predictor.rb

```diff
@@ -3,17 +3,28 @@
 module Crystalball
   # Class that predicts test failures with given execution map and sources diff
   class Predictor
-    attr_reader :map, :diff
+    attr_reader :map, :diff, :spec_pattern
 
-    def initialize(map, source_diff)
+    def initialize(map, source_diff, spec_pattern)
       @map = map
       @diff = source_diff
+      @spec_pattern = spec_pattern
     end
 
     def cases
+      cases_from_changed_code + cases_from_changed_specs
```

Comment:
I was thinking about it earlier. It feels like there should be an array of prediction patterns that you can pass to predictor. That will make it somewhat extendable.
Something like that:
```
p = Predictor.new(map, source_diff) # by default it has only one default "pattern" which is `cases_from_changed_code`
p.prediction_patterns << Patterns::NewSpecs.new(spec_regexp) # your "new_specs" pattern
p.prediction_patterns << Patterns::CorrespondingRailsFile.new # some other pattern
p.cases
```

Reply:
If we do it like that, I'd much rather not add any defaults and allow everything to be specified explicitly. Also how would we use it from `Crystallball.foresee`? Something like:
```ruby
Crystallball.foresee do |predictor|
  predictor.patterns << CrystallBall::Pattern::ChangedDependencies.new
  predictor.patterns << CrystallBall::Pattern::ChangedSpecFiles.new(spec_regexp)
end
```

Reply:
yeah something like that